### PR TITLE
Added model export to biolqm

### PR DESCRIPTION
**Pull request by vincent-noel**
MaBoSS has an option (-l) to dump logical rules of a model. 
While this is a complex problem, is most cases it should work just fine. And if it doesn't, it should return an error.

### CODE REVIEW

File: maboss/simulation.py [hunk 1]
```diff
@@ -336,3 +343,15 @@ def copy_and_update_parameters(sim, parameters):
     new_sim.update_parameters( **parameters )
     return new_sim
 
+def to_biolqm(maboss_model):
+
```
Comment (pauleve) [inline maboss/simulation.py]: you should use
```
biolqm = import_colomoto_tool("biolqm")
```

File: maboss/simulation.py [hunk 2]
```diff
@@ -17,6 +19,7 @@
 import subprocess
 import tempfile
 import shutil
+import biolqm
```
Comment (pauleve) [inline maboss/simulation.py]: Replace with
```
from colomoto_jupyter import import_colomoto_tool
```

File: maboss/simulation.py [hunk 3]
```diff
@@ -336,3 +343,15 @@ def copy_and_update_parameters(sim, parameters):
...
+    path = tempfile.mkdtemp()
+    bnet_fd, bnet_filename = tempfile.mkstemp(dir=path, suffix='.bnet')
+    os.close(bnet_fd)
+
+    with open(bnet_filename, "w") as bnet_file:
+        for node, rule in maboss_model.get_logical_rules().items():
+            bnet_file.write("%s, %s\n" % (node, rule))
+
+    return biolqm.load(bnet_filename)
+
```
Comment (pauleve) [inline maboss/simulation.py]: You can also do a similar thing to convert to minibn:
```
def to_minibn(maboss_model):
    from colomoto import minibn # minibn is not (yet) a proper colomoto tool
    ... output to .bnet file
    return minibn.BooleanNetwork.load(bnet_filename)
```

File: maboss/simulation.py [hunk 4]
```diff
@@ -336,3 +343,15 @@ def copy_and_update_parameters(sim, parameters):
     new_sim.update_parameters( **parameters )
     return new_sim
 
+def to_biolqm(maboss_model):
+    
+    path = tempfile.mkdtemp()
+    bnet_fd, bnet_filename = tempfile.mkstemp(dir=path, suffix='.bnet')
```
Comment (pauleve) [inline maboss/simulation.py]: You can use a helper function provided by colomoto-jupyter (which will automatically delete the file when the notebook stops):
```
bnet_filename = new_output_file("bnet")
```
you need
```
from colomoto_jupyter.sessionfiles import new_output_file
```
(see https://github.com/GINsim/GINsim-python/blob/master/biolqm/__init__.py for examples of model conversions)

File: setup.py [hunk 5]
```diff
@@ -19,4 +19,5 @@
         "ipywidgets",
         "matplotlib",
         "pandas",
+        "ginsim"
```
Comment (pauleve) [inline setup.py]: I would drop this requirement

File: conda/meta.yaml [hunk 6]
```diff
@@ -27,6 +27,7 @@ requirements:
     - python
     - matplotlib
     - ipywidgets
+    - ginsim-python
```
Comment (pauleve) [inline conda/meta.yaml]: We can avoid having a strong requirement on ginsim, see my comments below.

**Comment by pauleve** [comment]
You can also add corresponding menu entries in `maboss/widgets/menu.py`

File: maboss/simulation.py (hunk 1 above)
Comment (vincent-noel) [reply maboss/simulation.py]: Done

File: maboss/simulation.py (hunk 2 above)
Comment (vincent-noel) [reply maboss/simulation.py]: Done

File: maboss/simulation.py (hunk 3 above)
Comment (vincent-noel) [reply maboss/simulation.py]: Done

File: maboss/simulation.py (hunk 4 above)
Comment (vincent-noel) [reply maboss/simulation.py]: Done

File: maboss/simulation.py (hunk 1 above)
Comment (vincent-noel) [reply maboss/simulation.py]: Done

File: maboss/simulation.py (hunk 4 above)
Comment (vincent-noel) [reply maboss/simulation.py]: Done

File: conda/meta.yaml (hunk 6 above)
Comment (vincent-noel) [reply conda/meta.yaml]: Done

File: setup.py (hunk 5 above)
Comment (vincent-noel) [reply setup.py]: Done

**Comment by vincent-noel** [comment]
Not sure I understood about the dependencies : 

```
Traceback (most recent call last):
  File "/home/travis/build/vincent-noel/pyMaBoSS/test/test_conversion.py", line 23, in test_convert_reprod_all
    to_biolqm(model)
  File "/home/travis/build/vincent-noel/pyMaBoSS/maboss/simulation.py", line 349, in to_biolqm
    biolqm = import_colomoto_tool("biolqm")
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/colomoto_jupyter/__init__.py", line 58, in import_colomoto_tool
    mod = __import__(modname, globals(), locals())
ImportError: No module named biolqm
```

Do we need biolqm package, which is inside ginsim (in pypi) and ginsim-python in conda ? Or do we need to run something after install, like maboss_setup.py ?

```
Traceback (most recent call last):
  File "/home/travis/build/colomoto/pyMaBoSS/test/test_conversion.py", line 28, in test_convert_reprod_all_minibn
    minibn_model = to_minibn(model)
  File "/home/travis/build/colomoto/pyMaBoSS/maboss/simulation.py", line 358, in to_minibn
    from colomoto import minibn
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/colomoto/minibn.py", line 206
    super(MultiValuedNetwork, self).__init__(*args, **kwargs, Symbol_class=MVVar)
```

Not sure what to do about this one. In the test environment, I'm not even sure I'm installing the colomoto package.

**Comment by pauleve** [comment]
You still need ginsim and colomoto-jupyter conda packages in the test environment (but people will be able to use maboss without having to install ginsim if they don't use to_biolqm)

**Comment by vincent-noel** [comment]
These modifications seems to work now. 

However, I found two issues : 
- colomoto-jupyter and colomoto are not compatible with python2.7. I will have a look, usually it's very easy to fix, but if it's not, we should put a condition on python version in the dependencies.
- ginsim-python is not compatible with python 3.5 and 3.7 (but it's compatible with 2.7 ?!), according to conda install.

**Comment by pauleve** [comment]
Strange, ginsim-python should be compatible with python 3.7... actually the latest (dev) docker is on python 3.7.. what is the error when you try to install in a python 3.7 environment?

We decided to not support python 2.7, especially as its official support will be soon over.

**Comment by vincent-noel** [comment]
conda create -q -n test-environment -c colomoto -c potassco python=3.5 ginsim-python
Collecting package metadata: ...working... done
Solving environment: ...working... failed

UnsatisfiableError: The following specifications were found to be in conflict:
  - ginsim-python -> ginsim -> clingo -> python[version='>=3.6,<3.7.0a0'] -> openssl[version='>=1.1.1a,<1.1.2a']
  - python=3.5

Ça vient de clingo.

**Comment by vincent-noel** [comment]
I changed the channel for clingo from potassco to potassco/label/dev to make it work.

**Pull request merged**